framework/src/mesh/MooseMesh.C

```diff
@@ -3521,9 +3546,22 @@ MooseMesh::deleteRemoteElements()
 }
 
 void
-MooseMesh::cacheVarIndicesByFace(const std::vector<const MooseVariableFieldBase *> & moose_vars)
+MooseMesh::cacheFaceInfoVariableOwnership() const
 {
-  buildFiniteVolumeInfo();
+  std::vector<const MooseVariableFieldBase *> moose_vars;
+
+  for (const auto i : make_range(_app.feProblem().numNonlinearSystems()))
```

Comment:
```suggestion
  mooseAssert(!Threads::in_threads, "Performing writes to elem info dof indices. This must be done unthreaded");

  for (const auto i : make_range(_app.feProblem().numNonlinearSystems()))
```

Reply:
This is probably overly paranoid but I think of this whenever I see hard-coded THREAD_ID arguments